doc/contribute/index.rst

```diff
@@ -340,7 +340,8 @@ following exceptions:
 * Use C89-style single line comments, ``/*  */``. The C99-style single line
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
-
+* Use the ``u|s<size>_t`` types defined in ``include/zephyr/types.h`` instead of
+  the ``[u]int<size>_t`` standard types defined in ``stdint.h``.
```

Comment:
Part of review comments on PR, I was instructed not to even use default data type (e.g. int, unsigned int, etc) and use "zephyr types" (i.e. s32_t, u32_t). However, I don't think this should be in guideline. It is natural to use default data types.

This comment looks good to me in the sense that, at least, it provides clarity on using `u|s<size>_t` type instead of `[u]int<size>_t`. However, I hope this would not create confusion, because there are lot of places using `[u]int<size>_t` in zephyr currently.

Reply:
There is a reason why we ask everybody to use the zephyr types instead of the stdint.h ones, and it has to do with compatibility with multiple C libraries. I think this belongs here in this guide, because it's a deviation from the Linux coding standard which we use as a reference, and it is something that we want to enforce in Pull Requests.

Reply:
> it is something that we want to enforce in Pull Requests
I agree to enforce zephyr types `u|s<size>_t` instead of `[u]int<size>_t`.

However, it should not be enforced to use zephyr types instead of standard data types(unsigned int, int, etc) using which zephyr types are derived. 
I know above line does not convey that, but PR review should be done accordingly. For example, return types such as int, unsigned int, etc. should be allowed. It is very common to use int as return type, I don't see reason enforcing using s32_t instead.

Reply:
> I know above line does not convey that, but PR review should be done accordingly. For example, return types such as int, unsigned int, etc. should be allowed. It is very common to use int as return type, I don't see reason enforcing using s32_t instead.

Normal return types are used in multiple places in zephyr, their use is just fine. Not sure if that should be written somewhere.

Reply:
Well, I was just concerned by the comment I got on PR to "use s32_t, u32_t, etc everywhere". Only when using variables for specific purpose, this use should be enforced.